cmd/getStorageValue.go

```diff
@@ -0,0 +1,151 @@
+	"github.com/makerdao/vulcanizedb/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var getStorageValueBlockNumber int64
+
+// getStorageValueCmd represents the getStorageValue command
+var getStorageValueCmd = &cobra.Command{
+	Use:   "getStorageValue",
+	Short: "Gets all storage values for configured contracts at the given block.",
+	Long: `Fetches and persists storage values of the configured contracts at a given block. It is important to note that the storage value gotten with this
+	command may not be different from the previous block in the database.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		SubCommand = cmd.CalledAs()
+		LogWithCommand = *logrus.WithField("SubCommand", SubCommand)
+		fmt.Println("getStorageValue called")
+		return getStorageAt(getStorageValueBlockNumber)
+	},
+}
+
+func init() {
+	getStorageValueCmd.Flags().Int64VarP(&getStorageValueBlockNumber, "get-storage-value-block-number", "b", -1, "block number to fetch storage at for all configured transformers")
+	rootCmd.AddCommand(getStorageValueCmd)
+}
+
+func getStorageAt(blockNumber int64) error {
+	blockChain := getBlockChain()
+	db := utils.LoadPostgres(databaseConfig, blockChain.Node())
+	storageInitializers := exportTransformers()
+	commandRunner := NewStorageValueCommandRunner(blockChain, &db, storageInitializers, blockNumber)
+	return commandRunner.Run()
+}
+
+func NewStorageValueCommandRunner(bc core.BlockChain, db *postgres.DB, initializers []transformer.StorageTransformerInitializer, blockNumber int64) StorageValueCommandRunner {
+	return StorageValueCommandRunner{
+		bc:              bc,
+		db:              db,
+		headerRepo:      repositories.NewHeaderRepository(db),
+		storageDiffRepo: storage2.NewDiffRepository(db),
+		initializers:    initializers,
+		blockNumber:     blockNumber,
+	}
+}
+
+type StorageValueCommandRunner struct {
+	bc              core.BlockChain
+	db              *postgres.DB
+	headerRepo      repositories.HeaderRepository
+	storageDiffRepo storage2.DiffRepository
+	initializers    []transformer.StorageTransformerInitializer
+	blockNumber     int64
+}
+
+func (r *StorageValueCommandRunner) Run() error {
+	addressToKeys, getKeysErr := r.getStorageKeys()
+	if getKeysErr != nil {
+		return getKeysErr
+	}
+
+	header, getHeaderErr := r.headerRepo.GetHeader(r.blockNumber)
+	if getHeaderErr != nil {
+		return getHeaderErr
+	}
+
+	for address, keys := range addressToKeys {
+		persistStorageErr := r.getAndPersistStorageValues(address, keys, r.blockNumber, header.Hash)
+		if persistStorageErr != nil {
+			return persistStorageErr
+		}
+	}
+
+	return nil
+}
+
+func (r *StorageValueCommandRunner) getAndPersistStorageValues(address common.Address, keys []common.Hash, blockNumber int64, headerHash string) error {
+	blockNumberBigInt := big.NewInt(blockNumber)
+	for _, key := range keys {
+		value, getStorageErr := r.bc.GetStorageAt(address, key, blockNumberBigInt)
```

Comment:
Thinking that a future performance boost could be to batch these `GetStorageAt` calls.